src/d2r_image/ocr.py

```diff
@@ -0,0 +1,255 @@
+    # case: two 1's within a string; e.g., "SIIPER MANA POTION"
+    try:
+        text = II_U.sub('U', ocr_output)
+    except:
+        # logging.error(f"Error _II_ -> _U_ on {ocr_output}")
+        text = ocr_output
+    # case: two 1's within a string; e.g., "S11PER MANA POTION"
+    try:
+        text = ONEONE_U.sub('U', text)
+    except:
+        # logging.error(f"Error _11_ -> _U_ on {ocr_output}")
+        pass
+    # case: an I within a number or by a sign; e.g., "+32I to mana attack rating"
+    try:
+        text = I_1.sub('1', text)
+    except:
+        # logging.error(f"Error I -> 1 on {ocr_output}")
+        pass
+    # case: a 1 within a string; e.g., "W1RT'S LEG"
+    try:
+        text = ONE_I.sub('I', text)
+    except:
+        # logging.error(f"Error 1 -> I on {ocr_output}")
+        pass
+    # case: a solitary I; e.g., " I TO 5 DEFENSE"
+    cnt = 0
+    while True:
+        cnt += 1
+        if cnt > 30:
+            # logging.error(f"Error ' I ' -> ' 1 ' on {ocr_output}")
+            break
+        if " I " in text:
+            text = text.replace(" I ", " 1 ")
+            continue
+        elif ' I\n' in text:
+            text = text.replace(' I\n', ' 1\n')
+            continue
+        elif '\nI ' in text:
+            text = text.replace('\nI ', '\n1 ')
+            continue
+        break
+    # case: a solitary S; e.g., " 1 TO S DEFENSE"
+    cnt = 0
+    while True:
+        cnt += 1
+        if cnt > 30:
+            # logging.error(f"Error ' S ' -> ' 5 ' on {ocr_output}")
+            break
+        if " S " in text:
+            text = text.replace(" S ", " 5 ")
+            continue
+        elif ' I\n' in text:
+            text = text.replace(' S\n', ' 5\n')
+            continue
+        elif '\nI ' in text:
+            text = text.replace('\nS ', '\n5 ')
+            continue
+        break
+    # case: consecutive I's; e.g., "DEFENSE: II"
+    repeat = False
+    cnt = 0
+    while "II" in text:
+        cnt += 1
+        if cnt > 30:
+            # logging.error(f"Error 4 on {ocr_output}")
+            break
+        text = text.replace("II", "11")
+        repeat = True
+        repeat_count += 1
+    if repeat and repeat_count < 10:
+        _fix_regexps(text)
+    return text
+
+
+def _check_known_errors(text):
+    for key in ERROR_RESOLUTION_MAP:
+        if key in text:
+            text = text.replace(key, ERROR_RESOLUTION_MAP[key])
+    return text
+
+def _ocr_result_dictionary_check(
+    original_text: str,
+    confidences: list,
+    word_list: list = all_words(),
+    normalized_lev_threshold: float = 0.6,
+    ocr_confidence_threshold: float = 0.9
+    ) -> str:
+    confidences = [x/100 for x in confidences]
+    if all(x >= ocr_confidence_threshold for x in confidences):
+        return original_text
+    #print(f"{original_text} {confidences}")
+    text = original_text.replace('\n', ' NEWLINEHERE ')
+    corrected_text = ""
+    word_count = 0
+    for word in text.split(' '):
```

Comment:
Also re. splitting in general, and I'm not sure if you would actually want to do anything about this: it's kind of a shame that we can't take advantage of knowing the sort of phrases the game is likely to throw at us, aka the context.  If I as a player see "C_______ VOUGE" I have a pretty good idea of what that first word might be, even if I thought I saw a bunch of 5s instead of S's.  This is just something I was thinking about though, I think this way of doing things with the right settings seems like it could work totally fine without the context of where a word is.  Passing in some context could also help make `word_list` smaller in a way - even if you only knew the word before or after, it could drastically limit the candidate words.  If you could quickly look up all of the words that come before `wand` in various game items, for instance.  But yeah, probably this speculation I'm giving you is completely unnecessary and premature optimization is the root and all that